### ISSUE TITLE
Custom spec style

### ISSUE THREAD

**Issue by ocherniavskyi-lohika-tix** [kotest/kotest#2702]
Hi, i want to add custom spec style, nothing serious just rename FunSpec Methods like

container -> migration
test or preparation or validation 

e.g. 

```
class MyMigration: MigrationSpec(
{    
    migration("some migration name"){
       preparation("some name"){
           // test code
       }

      validation("some name"){
           // test code
       }
    }
}
)
```

is it possible to add support to IntelliJ IDEA plugin to get possibility to run this tests using right click?

**Comment by sksamuel** [comment]
Creating the spec is easy.
Updating the plugin is hard. We'd have to make it dynamic, so you could somehow configure which methods to look for.
You would need to add a settings feature to the plugin so those names were persistent.
I'm not sure its worth it. 

What about just adding migration: to the test name? If you use free spec then its pretty open ended.

**Comment by ocherniavskyi-lohika-tix** [comment]
@sksamuel unfortunately i need to add some configuration to this methods, smal but it will become boilerplate on big amount of tests 
If there is a possibility to add settings file could you point me to doc or example how to do it?

**Comment by Jadarma** [comment]
@sksamuel Indeed, supporting individual sub-context and test runs in a dynamic fashion would be complicated, but is it a reasonable compromise to make the plugin display the "play" gutter icon for any class that extends `io.kotest.core.spec.Spec`?
That would allow users to make their own specific DSLs without introducing too much complexity in the plugin, and it would be a great QoL because you don't need to create the run configuration by hand, at the cost of only being able to run the entire spec.

**Comment by sksamuel** [comment]
I think the easiest thing to do would be have a settings area in the plugin that accepted a list of names which it would look for as "tests". That is basically how fun spec, should spec, etc work. 

1. Create a settings area for the plugin so that users can bring it up in intellij -> settings -> kotest
2. Update the PSI logic to use the settings value when looking for "tests".

Someone would need to pick up this work if they want the feature, and it would mean you need to learn about how intellij plugins work. I would be able to help guide you in the right direction.

**Comment by sksamuel** [comment]
Once this is merged, it will promote a public API for creating custom spec styles with full IDE support.
https://github.com/kotest/kotest/pull/5814
